services/grafana/client.go

```diff
@@ -297,54 +296,37 @@ func (c *Client) CreateAnnotation(ctx context.Context, tags []string, text strin
 		return "", errors.Wrap(err, "failed to marhal request")
 	}
 
-	u := url.URL{
-		Scheme: "http",
-		Host:   c.addr,
-		Path:   "/api/annotations",
-	}
-
-	// TODO should be updated to use c.do
-
-	resp, err := c.http.Post(u.String(), "application/json", &buf)
-	if err != nil {
-		return "", errors.Wrap(err, "failed to make request")
-	}
-	defer resp.Body.Close() //nolint:errcheck
+	var headers = make(http.Header)
+	headers.Add("Authorization", authorization)
 
 	var response struct {
 		Message string `json:"message"`
 	}
-	if err = json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return "", errors.Wrap(err, "failed to decode JSON response")
+
+	if err := c.do(ctx, "POST", "/api/annotations", headers, buf.Bytes(), &response); err != nil {
+		logrus.Errorln(err)
+		return "", errors.Wrap(err, "failed to do request")
```

Comment:
Does this add new information to the error? Maybe it's better to replace by "failed to send request to grafana"?

Reply:
Yeah, changed to another.